packages/web-components/src/components/va-search/va-search.tsx

```diff
@@ -0,0 +1,44 @@
+import { Component, Host, h, Prop, Event, EventEmitter } from '@stencil/core';
+
+@Component({
+  tag: 'va-search',
+  styleUrl: 'va-search.css',
+  shadow: true,
+})
+export class VaSearch {
+  @Prop() action: string;
+
+  @Prop() label: string = 'Search';
+
+  @Prop() method?: 'GET' | 'POST' | 'dialog';
+
+  @Event({ bubbles: true, composed: true })
+  submitEvent: EventEmitter;
+
+  private handleSubmitEvent(event: KeyboardEvent | MouseEvent) {
+    event.preventDefault();
+    this.submitEvent.emit(event);
+  }
```

Comment:
I'm trying this in Storybook, but the function passed in isn't being fired.

Reply:
That's my bad - I got `bubbles` and `composed` confused again. According to [some documentation](https://javascript.info/shadow-dom-events#event-composed):

> Most events successfully bubble through a shadow DOM boundary. There are few events that do not.
>
> This is governed by the `composed` event object property.

Since `composed` isn't true for the `submit` event that the DOM fires, we may have to [create and fire](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events) a synthetic [`SubmitEvent`](https://developer.mozilla.org/en-US/docs/Web/API/SubmitEvent/SubmitEvent) if we want to rely on the native `onSubmit` handler.